packages-exp/messaging-exp/src/index.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  ComponentContainer,
+  ComponentType,
+  InstanceFactory
+} from '@firebase/component';
+import { ERROR_FACTORY, ErrorCode } from './util/errors';
+
+import { FirebaseMessaging } from '@firebase/messaging-types-exp';
+import { MessagingService } from './messaging-service';
+import { _FirebaseNamespace } from '@firebase/app-types/private';
+import { _registerComponent } from '@firebase/app-exp';
+import { isSupported } from './helpers/isSupported';
+
```

Comment:
```suggestion
export * from './api';
```

Reply:
Perhaps we should create 2 entry points, one for the client and one for the service worker. Then we only export the applicable APIs from them respectively.